repositories/translationRepository.ts

```diff
@@ -1,12 +1,19 @@
 import LocaleString from "../entities/LocaleString";
 
+const locale_cache = new Map();
+
 async function getTranslation(
   locale: LocaleString
 ): Promise<Record<string, string>> {
+    
+  if (typeof window !== "undefined" && locale_cache.has(locale)) {
+    return locale_cache.get(locale)
+  } 
```

Comment:
This block is supposed to be inside the next `if (typeof window !== "undefined") { ... }`.

```typescript
if (typeof window !== "undefined") {
    if (localeCache.has(locale)) {
        return localeCache.get(locale)
    }

    const response = await fetch(`/static/translation/${locale}.json`); 
    // ...
}
```